netaddr/ip/__init__.py

```diff
@@ -1683,28 +1683,33 @@ def spanning_cidr(ip_addrs):
 
     :return: a single spanning `IPNetwork` subnet.
     """
-    sorted_ips = sorted(
-        [IPNetwork(ip) for ip in ip_addrs])
-
-    if not len(sorted_ips) > 1:
+    ip_addrs_count = 0
+    min_network = None
+    max_network = None
+    for ip in ip_addrs:
+        ip_addrs_count += 1
+        network = IPNetwork(ip)
+        if min_network is None or network < min_network:
+            min_network = network
+        if max_network is None or network > max_network:
+            max_network = network
+
+    if not ip_addrs_count > 1:
```

Comment:
I only count the elements in order to make this check. I don't know if it's really necessary, or if what we really want to check is that `min_network != max_network`, but this way we preserve the current behavior, which is probably the safest.